src/Microsoft.Tye.Extensions.Configuration/TyeConfigurationExtensions.cs

```diff
@@ -0,0 +1,71 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.Extensions.Configuration
+{
+    public static class TyeConfigurationExtensions
+    {
+        public static Uri? GetServiceUri(this IConfiguration configuration, string name, string? binding = null)
+        {
+            var key = GetKey(name, binding);
+
+            var connectionString = configuration[$"connectionstrings:{key}"];
+            if (!string.IsNullOrEmpty(connectionString))
```

Comment:
Is this weird? Feels weird.

Reply:
which part? 

I haven't written the docs yet, but here's what I was going to explain.

> Use the method (`GetConnectionString`, `GetServiceUri`, `GetServiceHost`) **that best describes the kind of config/secret you need to access**.

We need **something** that does this, because in the deployment flow we don't know what code you wrote in your app. Ex: the configuration used to access redis is commonly a hostname/port in development, and commonly a connection string (arbitrary text) in production.

To me it doesn't seem weird once you have the realization that:

- We don't/can't know what code is in your app
- We don't know how to decompose arbitrary text (secrets) into host/port/protocol

---

The potential for mistakes does exist here btw, because imagine that you used `GetServiceHost` for redis. That won't work in production. So we probably need more guidance.

Reply:
We had a discussion about this more in teams, and ended up with this as the best proposal:

- Don't allow connection strings to override GetServiceUri
- Get rid of GetServiceHost (use GetConnectionString instead for things like redis)
- In dev, we populate CONNECTIONSTRING_REDIS to be the host:port (prepend the protocol if specified)